### PR TITLE
CDSR-1995

### PR DESCRIPTION
**Pull request by airudah2000**

### CODE REVIEW

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaim.models.claim
+
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.cdsreimbursementclaim.models.eis.declaration.response.BankAccountDetails
+import uk.gov.hmrc.cdsreimbursementclaim.models.ids.MRN
+import uk.gov.hmrc.cdsreimbursementclaim.utils.MapFormat
+
+final case class ScheduledOverpaymentsClaim(
```
Comment (arturopala) [inline app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala:24]: This has to be the same as `OverpaymentsScheduledJourney.Output` in the frontend

**Review by arturopala — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala (hunk 1 above)
Comment (airudah2000) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala:24]: @arturopala, I have updated the schema to match output from frontend after pairing with @mghmay

Comment (arturopala) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala:24]: @airudah2000 @mghmay I think the container for reimbursement value should be AmountPaidWithCorrect in overpayments

Comment (arturopala) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala:24]: And it should be interpreted accordingly

**Review by arturopala — APPROVED** [review]

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  scheduledDocument: EvidenceDocument,
+  supportingEvidences: Seq[EvidenceDocument]
+) {
+  lazy val combinedReimbursementClaims: Map[TaxCode, AmountPaidWithCorrect] =
+    reimbursementClaims.values.reduceOption((x, y) => x |+| y).getOrElse(Map.empty)
+
+  def documents: Seq[EvidenceDocument] = scheduledDocument +: supportingEvidences
+
+  def totalReimbursementAmount: BigDecimal =
+    reimbursementClaims.values.map(_.values.map(_.correctAmount).sum).sum
```
[the remark this answers is not in the document]
Comment (airudah2000) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/ScheduledOverpaymentsClaim.scala]: Updated

**Pull request merged**